### PR TITLE
test: add ramda placeholder and make tests consistent

### PR DESCRIPTION
**Pull request by js636f**
Batch 6

Files:
test/
- isNotRegExp.js
- isNotString.js
- isNotUndefined.js
- isNotValidDate.js

src/
- isValidDate.js (curry1)

### CODE REVIEW

File: test/isNotUndefined.js [hunk 1]
```diff
@@ -1,15 +1,34 @@
...
-  it('tests a value for complement of `undefined`', function() {
-    assert.isFalse(RA.isNotUndefined(void 0));
-    assert.isFalse(RA.isNotUndefined(undefined));
-    assert.isTrue(RA.isNotUndefined(null));
-    assert.isTrue(RA.isNotUndefined([]));
-    assert.isTrue(RA.isNotUndefined({}));
-    assert.isTrue(RA.isNotUndefined(0));
-    assert.isTrue(RA.isNotUndefined(''));
+  context('given a non-undefined value', function() {
+    specify('it should return true', function() {
```
Comment (char0n) [inline test/isNotUndefined.js]:
```suggestion
    specify('should return true', function() {
```

File: test/isNotUndefined.js [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+      assert.isTrue(RA.isNotUndefined(''));
+
+      if (Symbol !== 'undefined') {
+        assert.isTrue(RA.isNotUndefined(Symbol));
+      }
+    });
+  });
+
+  context('given an undefined value', function() {
+    specify('it should return false', function() {
```
Comment (char0n) [inline test/isNotUndefined.js]:
```suggestion
    specify('should return false', function() {
```

File: test/isNotValidDate.js [hunk 3]
```diff
@@ -1,37 +1,50 @@
...
-    assert.isTrue(RA.isNotValidDate(Object('a')));
-
-    if (Symbol !== 'undefined') {
-      assert.isTrue(RA.isNotValidDate(Symbol));
-    }
-
-    assert.isTrue(RA.isNotValidDate(null));
-    assert.isTrue(RA.isNotValidDate(undefined));
+  context('given a non-date value', function() {
+    specify('it should return true', function() {
```
Comment (char0n) [inline test/isNotValidDate.js]:
```suggestion
    specify('should return true', function() {
```

File: test/isNotValidDate.js [hunk 4]
```diff
@@ -1,37 +1,50 @@
...
+        assert.isTrue(RA.isNotValidDate(Symbol));
+      }
+
+      assert.isTrue(RA.isNotValidDate(null));
+      assert.isTrue(RA.isNotValidDate(undefined));
+    });
+  });
+
+  context('given a date value', function() {
+    specify('it should return false', function() {
```
Comment (char0n) [inline test/isNotValidDate.js]:
```suggestion
    specify('should return false', function() {
```

**Comment by js636f** [comment]
Done!

**Pull request merged**